src/PointwiseFunctions/GeneralRelativity/WeylMagnetic.cpp

```diff
@@ -0,0 +1,139 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/WeylMagnetic.hpp"
+
+#include "DataStructures/LeviCivitaIterator.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/VectorImpl.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+///\cond
+namespace gr {
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ii<DataType, SpatialDim, Frame> weyl_magnetic(
+    const tnsr::ijj<DataType, SpatialDim, Frame>& grad_extrinsic_curvature,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric) noexcept {
+  tnsr::ii<DataType, SpatialDim, Frame> weyl_magnetic_part{};
+  weyl_magnetic<SpatialDim, Frame, DataType>(make_not_null(&weyl_magnetic_part),
+                                             grad_extrinsic_curvature,
+                                             spatial_metric);
+
+  return weyl_magnetic_part;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void weyl_magnetic(
+    const gsl::not_null<tnsr::ii<DataType, SpatialDim, Frame>*>
+        weyl_magnetic_part,
```

Comment:
use the helper function `destructive_resize_components` to resize the return-by-pointer only if necessary